sendgrid_test.go

```diff
@@ -81,6 +83,41 @@ func TestGetRequestSubuser(t *testing.T) {
 	ShouldHaveHeaders(&request, t)
 }
 
+func Test_test_set_residency_eu(t *testing.T) {
```

Comment:
take a look at test cases here https://github.com/sendgrid/sendgrid-java/pull/743/files#diff-901f4afdd8c5674fd44b77810400815a6539c341bc83a051d9843c21a28fd966, whichever is set latest, that decides the host.

Reply:
In go, we do not have the option of setting host on our own, that's why I didn't add this test